tracks/elm/exercises/leap/mentoring.md

```diff
@@ -0,0 +1,90 @@
+This is a good exercise to discuss the describability of logic, as most students solve the problem using a conditional statement that is hard to describe in language, and there are other options.
+
+This exercise introduces `Basics.remainderBy`, the `let` / `in` syntax and partial application / currying.
+
+### Reasonable solutions
+
+The easiest / most common solution probably involves partially applying `Basics.remainderBy` and using a slightly complicated if statement:
+
+```elm
+isLeapYear : Int -> Bool
+isLeapYear year =
+    let
+        divisibleBy number = 
+            Basics.remainderBy number year == 0         
+    in
+        (
+            divisibleBy 4 && 
+            not (divisibleBy 100)
+        )
+        || divisibleBy 400
+```
+
+The complicated if statement is a bit of a problem, as it is hard to describe in language, and hence talking about the code with others is difficult.
+
+Using a case statement turns the if statement in to a truth table, and is probably a bit clearer, but at the expense of some extra code:
+
+```elm
+isLeapYear : Int -> Bool
+isLeapYear year =
+    let
+        divisibleBy number = 
+            Basics.remainderBy number year == 0         
+    in
+		case ( divisibleBy 4, divisibleBy 100, divisibleBy 400 ) of
+			( True, True, True ) ->
+				True
+
+			( True, True, False ) ->
+				False
+
+			( True, False, False ) ->
+				True
+
+			_ ->
+				False
+```
+
+If you think about the easiest way to describe the logic to somebody else, you probably come up with a solution like this:
+
+```elm
+isLeapYear : Int -> Bool
+isLeapYear year =
+    let
+        divisibleBy number = 
+            Basics.remainderBy number year == 0         
+    in
+		if divisibleBy 400 then
+			True
+
+		else if divisibleBy 100 then
+			False
+
+		else if divisibleBy 4 then
+			True
+
+		else
+			False
```

Comment:
The following snippet is one that I give as feedback on the Haskell track:

> Don't write `if p then True else False`. This is equivalent to simply `p`. Similarly, don't write `if p then False else True`. This is equivalent to simply `not p`. You can combine `p` and `not p` with `&&` and `||` for your values of `p`.

On the other hand, there is one particular if-then-else solution that I do find readable (and not verbose):

```haskell
isLeapYear :: Integer -> Bool
isLeapYear year =
  if divisibleBy 100
  then divisibleBy 400
  else divisibleBy 4
  where
    isDivisibleBy n = year `rem` n == 0
```

Reply:
Hi Simon. I see your point here, but I'm afraid I don't agree with it. I am happy to have a little extra code to improve readability / clarity. Maybe there is a difference between the average elm (a beginner / easy to learn functional language) programmer and the average Haskell (a much more complete functional language) programmer.

I like the clarity of `if divisibleBy 400 then True`, and it is easy to describe in words. It takes me longer to work out what actually happens when reading your example.

So all in all I am afraid that I am not going to action this point.